synapse/storage/databases/main/events.py

```diff
@@ -617,7 +639,7 @@ def _persist_event_auth_chain_txn(
             events_to_calc_chain_id_for, event_to_auth_chain
         ):
             existing_chain_id = None
-            for auth_id in event_to_auth_chain[event_id]:
+            for auth_id in event_to_auth_chain.get(event_id, []):
```

Comment:
Is this a bugfix or is the code used differently due to the background update? (Same for the changes below in this file.)

Reply:
The latter, the BG update will no longer add empty entries for create events to the `event_to_auth_chain` map